biogrid/apps/biogrid-mvp/src/app/pages/input-page/input-page.tsx

```diff
@@ -4,15 +4,23 @@ import './input-page.scss';
 import DatePicker from 'react-datepicker';
 import { Client } from '../../client';
 import { useHistory } from 'react-router-dom';
-import Visual from "./BioGridVisual.jpg";
+import Visual from './BioGridVisual.jpg';
 
-function useInput(opts: { type: string }) {
-  const [value, setValue] = useState('');
+function useInput<T>(opts: { default: T }) {
+  const [value, setValue] = useState(opts.default);
+  function valueToType(value: string): T {
+    if (typeof opts.default === 'number') {
+      return (parseInt(value) as unknown) as T;
+    } else if (typeof opts.default === 'boolean') {
+      return (Boolean(value) as unknown) as T;
+    }
+    return (JSON.stringify(value) as unknown) as T;
```

Comment:
Can you shorten this to `return JSON.stringify(value) as T;`?

Reply:
It has to go through unkown or any first before it can be cast as T

Reply:
but if you're doing parseInt, don't you know that it will be a number?